my-app/src/AddBox/AddBox.jsx

```diff
@@ -0,0 +1,180 @@
+
+const getSuggestionValue = (object) => {
+  if (object.hasOwnProperty('label')) {
+    return object.label;
+  } else if (object.hasOwnProperty('name')) {
+    return object.name;
+  }
+}
+
+const getSuggestions = (value, options) => {
+  const escapedValue = escapeRegexCharacters(value.trim());
+  if (escapedValue === '') {
+    return [];
+  }
+  const regex = new RegExp('^' + escapedValue, 'i');
+  return options.filter(tag_or_person => regex.test(getSuggestionValue(tag_or_person)));
+};
+
+
+const renderSuggestion = (suggestion, {query}) => {
+  var img = 'none';
+  if (suggestion.hasOwnProperty('label')) {
+    img = 'TAG';
+  } else if (suggestion.hasOwnProperty('name')) {
+    img = 'PERSON';
+  }
+  return(
+         //TODO: how do you get theme classes in?
+    <span className='sugestionContent'>{getSuggestionValue(suggestion) + ' ' + img}</span>
+    );
+}
+
+
+export default class AddBox extends Component {
+  constructor() {
+    super();
+    this.state = {
+      value: '',
+      suggestions: [],
+      tags:[],
+    }
+  }
+
+  componentWillMount = value => {
+    var options = uniqLabelFast(this.props.tags);
+    this.setState({
+      options: options
+    });
+  };
+
+  componentWillReceiveProps = new_props => {
+    // Update lists of things when props change
+    var options = uniqLabelFast(new_props.tags);
+    this.setState({
+      options: options
+    });
+  };
+
+  onChange = (event, { newValue, method }) => {
+    this.setState({
+      value: newValue
+    });
+  };
+
+  onSuggestionsFetchRequested = ({ value }) => {
+    this.setState({
+      suggestions: getSuggestions(value, this.state.options)
+    });
+  };
+
+  onSuggestionsClearRequested = () => {
+    this.setState({
+      suggestions: []
+    });
+  };
+
+  handleTagSelection = (tag) => {
+    this.props.addTagToPerson(tag.label);
+  }
+
+  handlePersonSelection = (person) => {
+     this.props.setPerson(person);
+  }
+
+  onSuggestionSelected = (event, { suggestion }) => {
+    /**
+     *Handle existing selection
+    */
+    // determine whether suggestion is a person or a tag
+    if (suggestion.hasOwnProperty('label')) {
+      // handle selected tag
+      this.handleTagSelection(suggestion);
+    } else if (suggestion.hasOwnProperty('name')) {
+      // handle selected person
+      this.handlePersonSelection(suggestion);
+    }
+    this.setState({value:''});   
+  };
+  
+   _handleKeyPress = e => {
+    if (this.state.suggestions.length === 0 && this.state.value !== '') {
+          // Handles comlete entries
+      if (e.key === 'Enter') {
+        this.props.addThing(this.state.value);
+        this.setState({value:''});
+      }
+    }
+  };
+  
+  render() {
+    const { value, suggestions } = this.state;
+    const inputProps = {
+      placeholder: "Add Tags!",
+      value,
+      onChange: this.onChange
+    };
+
+    const renderInputComponent = inputProps => {
```

Comment:
I'm surprised to see a function that renders a component being passed into a child component as a prop.  This might work, but I don't think it's idiomatic in the React ecosystem.  It think it might be better to pass these values (value, suggestions) directly into the Autosuggest component as props, and it can pass them on through to another sub-component with a name.